src/components/ActionModal/ActionModal.tsx

```diff
@@ -0,0 +1,105 @@
+import { Button, Modal, ModalProps, Text, TextContent, TextVariants } from '@patternfly/react-core';
+import { TimesIcon } from '@patternfly/react-icons';
+import { css } from '@patternfly/react-styles';
+import { PropsWithChildren, ReactElement } from 'react';
+
+import { useRefresh } from 'hooks/useRefresh';
+import { DataValues, IServiceContainer } from 'hooks/useServiceContainer';
+
+import { ServiceContainerCreatingUpdating } from 'components/ServiceContainers/ServiceContainerCreatingUpdating';
+import { ServiceContainerLoading } from 'components/ServiceContainers/ServiceContainerLoading';
+
+import styles from './ActionModal.module.css';
+
+interface IActionModalProps {
+  modalTitle: string;
+  actionTitle: string;
+  isOpen: boolean;
+  isConfirmActionDisabled: boolean;
+  onToggle: () => void;
+  onSubmit: () => void;
+  serviceContainerCreatingUpdating: IServiceContainer;
+  modalVariant: ModalProps['variant'];
+  onSuccessActions?: ReactElement[];
+}
+
+/**
+ * Modal used for the creating or updating action.
+ *
+ * @param modalTitle - title of the modal
+ * @param actionTitle - title of the confirm action button
+ * @param isOpen - is the modal open?
+ * @param isConfirmActionDisabled - is the confirm action button disabled?
+ * @param onToggle - function toggling the modal visibility
+ * @param onSubmit - confirm action button callback
+ * @param serviceContainerCreatingUpdating - service container of the confirm action
+ * @param modalVariant - modal size variant
+ * @param onSuccessActions - additional action buttons to be displayed after susccessful confirm action
+ */
+export const ActionModal = ({
+  children,
+  modalTitle,
+  actionTitle,
+  isOpen,
+  isConfirmActionDisabled,
+  onToggle,
+  onSubmit,
+  serviceContainerCreatingUpdating,
+  modalVariant,
```

Comment:
The only option to apply the default value is the `modalVariant` parameter, but I don't see any value that would be good as 'default'. For example, on 'Mark Milestone as current' modal, this will be `small`, while on 'Artifact edit quality', it is `large`.

Reply:
done